src/main/java/net/datafaker/idnumbers/SwedenIdNumber.java

```diff
@@ -39,12 +39,16 @@ public PersonIdNumber generateValid(BaseProviders f, IdNumber.IdNumberRequest re
         LocalDate birthday = birthday(f, request);
         String end = generateEndPart(f);
         String basePart = DATE_TIME_FORMATTER.format(birthday)
-            + f.options().option(PLUS_MINUS)
+            + generateSymbol(birthday)
```

Comment:
Is the initial problem really solved here?
Now you generate the right symbol (`-` or `+`), but the date still may be `**0229+*****`.

Reply:
It's nothing incorrect with **0229+****, because 29th of February happend in few years. The incorrect value was 000229+****, because there was no 29th of February in 1900.

Reply:
1. Method `isYearOver100YearsAgo` expects year number in the first argument. But it's named `date` (not `year`), and you pass `dateString` in line 120 (`isYearOver100YearsAgo(dateString, ...`).

2. 
This code:
```java
    @RepeatedTest(1000)
    void swedishSsn_containsPlus_forPersonsOlderThan100Years() {
        PersonIdNumber person = new SwedenIdNumber().generateValid(new Faker(), new IdNumber.IdNumberRequest(123, 125, ANY));
        assertThat(person.idNumber()).contains("+");
        assertThat(SwedenIdNumber.isValidSwedishSsn(person.idNumber())).as(person.idNumber()).isTrue();
    }
```

still can generate invalid IDs, e.g. `991207+4219`, `991224+0851` or `991207+5430`.

Reply:
Fixed it!

also 991207+4219, 991224+0851 and 991207+5430 they are valid SSNs. For ssn validation logic I recommend using https://swedish.identityinfo.net/personalidentitynumber/validate?number=991207%2B4219